src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6718,7 +6718,13 @@ syntax switch
                         (ParameterSymbol? parameter, TypeWithAnnotations parameterType, FlowAnalysisAnnotations parameterAnnotations, bool isExpandedParamsArgument) =
                             GetCorrespondingParameter(i, parametersOpt, argsToParamsOpt, expanded, ref paramsIterationType);
 
-                        if (parameter is null)
+                        if (// This is known to happen for certain error scenarios, because
+                            // the parameter matching logic above is not as flexible as the one we use in `Binder.BuildArgumentsForErrorRecovery`
+                            // so we may end up with a pending conversion completion for an argument apparently without a corresponding parameter.
+                            parameter is null ||
+                            // In error recovery with named arguments, target-typing cannot work as we can get a different parameter type
+                            // from our GetCorrespondingParameter logic than Binder.BuildArgumentsForErrorRecovery does.
+                            node is BoundCall { HasErrors: true, ArgumentNamesOpt.IsDefaultOrEmpty: false, ArgsToParamsOpt.IsDefault: true })
```

Comment:
Could you explain why we want ArgsToParamsOpt to be default in order to enter this path?

Reply:
If it's not default it means we know the args-to-params mapping so we can continue. The problem is only in error scenarios where the bad BoundCall does not contain the mapping *even though* there are named arguments in the call. That's when we want to enter this path to skip further processing.

Reply:
I'm going to merge before the snap. Will follow up if you have more comments. Thanks.